components/ca-form.js

```diff
@@ -0,0 +1,1521 @@
+
+                // apply session stored form data if it exists
+                if (this.persist && window.sessionStorage && sessionStorage[this.src]) {
+
+                    const formData = sessionStorage[this.src] || '';
+
+                    if (formData !== '') {
+                        this.populate(JSON.parse(formData));
+                    }
+                }
+            }).catch(err => {
+                // eslint-disable-next-line
+                alert(`Schema not found! ${err}`);
+            });
+        }
+
+        /**
+         * @description Builds the HTML form based on the value of the assigned JSON .schema object
+         * @returns {magic} tricks.
+        */
+        render() {
+
+            // build table headers from schema
+            if (this._schema) {
+
+                // erase current form
+                clearElement(this.form);
+
+                // grab a handle to elements for rendering
+                const schema = this.selectedSchema;
+                let itemIndex = 1;
+                let group = '';
+                const schemaTitle = this._title || schema.title || '';
+                const schemaDesc = this._description || (Array.isArray(schema.description) ? schema.description.join('') : schema.description || '');
+
+                // insert title if it has one
+                if (schemaTitle !== '') {
+                    createElementLegacy(this.form, 'h2', { 'data-schema-title': schemaTitle }, null, schemaTitle);
+                }
+
+                if (schemaDesc !== '') {
+                    createElementLegacy(this.form, 'div', { class: 'ca-form-desc', 'data-schema-description': schemaDesc }, null, schemaDesc);
+                }
+
+                // add the selected schema to the form element
+                this.form.setAttribute('data-schema-type', schema.title);
+
+                // add schema option if more than one exist
+                if (this.schema.length > 1) {
+
+                    // create a form label container (acts as a row)
+                    const lbl = createElementLegacy(null, 'label', { for: 'schemaType', 'data-key': 'schemaType' });
+
+                    // create form field name
+                    createElementLegacy(lbl, 'span', null, 'Type');
+
+                    const select = createElementLegacy(lbl, 'select', { name: 'schemaType', id: 'schemaType' });
+
+                    for (let i = 0, l = this.schema.length; i < l; i++) {
+                        const opt = createElementLegacy(select, 'option', { value: this.schema[i].title }, this.schema[i].title);
+
+                        if (i === this.schemaIndex) {
+                            opt.selected = true;
+                        }
+                    }
+
+                    // switch schemas on schema type change
+                    select.onchange = () => {
+                        this.schemaIndex = this.selectedIndex;
```

Comment:
Good spot. But I think it does need to be an arrow function as we need access to "this" i.e. the component for this.schemaIndex and this.render
I've updated this.selectedIndex to select.selectedIndex
I think we should be good.